Backend/index.js

```diff
@@ -43,19 +43,25 @@ app.use(
     methods: ["GET", "POST", "DELETE", "PUT"],
   })
 );
-
 io.on("connection", (socket) => {
   const userId = socket.handshake.query.userId;
-  // Mark user as online
-  socket.join(userId);
+
+  if (!userId) {
+    socket.disconnect(true);
+    return;
+  }
+  socket.userId = userId;
+
   socket.on("disconnect", () => {
-    socket.leave(userId);
+    delete socket.userId;
```

Comment:
how is this different than socket.leave?

Reply:
My user was not disconnecting properly when they logged out, so I did some research and this was one of the things I found to try out and it worked. From what I learnt, delete ensures that the userId property is removed from the socket object when the user disconnects while leave ensures that the socket leaves the room identified by userId, so it will no longer receive events emitted to that room.